prow/tide/tide_test.go

```diff
@@ -1301,17 +1307,19 @@ func TestTakeAction(t *testing.T) {
 			t.Errorf("Wrong action. Got %v, wanted %v.", act, tc.action)
 		}
 
-		numCreated := 0
+		prowJobs := &prowapi.ProwJobList{}
+		if err := client.List(context.Background(), prowJobs); err != nil {
+			t.Fatalf("failed to list ProwJobs: %v", err)
+		}
+		numCreated := len(prowJobs.Items)
+
 		var batchJobs []*prowapi.ProwJob
-		for _, action := range fakeProwJobClient.Actions() {
-			switch action := action.(type) {
-			case clienttesting.CreateActionImpl:
-				numCreated++
-				if prowJob, ok := action.Object.(*prowapi.ProwJob); ok && prowJob.Spec.Type == prowapi.BatchJob {
-					batchJobs = append(batchJobs, prowJob)
-				}
+		for _, pj := range prowJobs.Items {
```

Comment:
Technically this isn't checking that they were created, right? Just that they exist in the end? So you could write a bad test where you accidentally add one from the start and not know?

Reply:
Yes, but I imagine doing that by accident is pretty hard, as the current table-driven design would then do that for all testcases

Reply:
Does the controller-runtime fake even allow for checking the actions?

Reply:
Does the controller-runtime fake even allow for checking the actions?

Reply:
Not to my knowledge, no.

Reply:
Guess we'll just have to make do with this then